sympy/solvers/ode.py

```diff
@@ -290,10 +290,10 @@
     "separable_reduced",
     "1st_power_series",
     "lie_group",
-    "nth_linear_constant_coeff_homogeneous",
-    "nth_linear_euler_eq_homogeneous",
     "nth_linear_constant_coeff_undetermined_coefficients",
     "nth_linear_euler_eq_nonhomogeneous_undetermined_coefficients",
+    "nth_linear_constant_coeff_homogeneous",
+    "nth_linear_euler_eq_homogeneous",
```

Comment:
The change you've made here probably does nothing, because the `homogeneous` and `undetermined_coefficients` hints are mutually exclusive. I meant to move them higher on the global list of hints, near the top.